pkg/controllermanager/controller/cloudprofile/machineversion/machine_image_version.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package machineversion
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/equality"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	controllermgrconfig "github.com/gardener/gardener/pkg/controllermanager/apis/config"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/cloudprofile/helper"
+)
+
+// ReconcileMachineImageVersions reconciles the versions of each machine image in the CloudProfile
+// returns true if the CloudProfile has to be updated, the updated CloudProfile or an error
+func ReconcileMachineImageVersions(logger *logrus.Entry, config *controllermgrconfig.CloudProfileControllerConfiguration, cloudProfile *gardencorev1beta1.CloudProfile) (bool, *gardencorev1beta1.CloudProfile, error) {
```

Comment:
I don't see the benefit of returning `bool, *gardencorev1beta1.CloudProfile`
- `*gardencorev1beta1.CloudProfile`: You anyway reflect the changes back in the passed `cloudProfile`
- `bool`: Let the caller check if the `cloudProfile` has been changed.